### ISSUE TITLE
Filter for ast.Search containing S or s does not match record containing U+017F (ſ) LATIN SMALL LETTER LONG S

### ISSUE THREAD

**Issue by nwt** [brimsec/zq#1207]
These three Unicode code points are equivalent under Unicode simple case folding:
* U+0053 (S) LATIN CAPITAL LETTER S
* U+0073 (s) LATIN SMALL LETTER S
* U+017F (ſ) LATIN SMALL LETTER LONG S

ast.Search ostensibly compares its Text field against record fields under simple case folding, so a search for one of those code points against a record field containing one of them should match. However, a search for S or s against a record field containing U+017F (ſ) LATIN SMALL LETTER LONG S does not match.

In other words, the following command should produce output but does not.
```
zq -t s - <<EOF
#0:record[a:string]
0:[\u017F;]
EOF
```

Fixing this requires changes to the behavior of filter.stringSearch and filter.stringCaseFinder.

U+017F (ſ) LATIN SMALL LETTER LONG S is archaic, so most users are unlikely to encounter this bug.
